airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3BaseChecks.java

```diff
@@ -77,6 +77,20 @@ public static void testMultipartUpload(final AmazonS3 s3Client, final String buc
     LOGGER.info("Finished verification for multipart upload mode");
   }
 
+  /**
+   * Checks that S3 custom endpoint uses a variant that only uses HTTPS
+   *
+   * @param endpoint URL string representing an accessible S3 bucket
+   */
+  public static boolean testCustomEndpointSecured(final String endpoint) {
+    // if user does not use a custom endpoint, do not fail
+    if (endpoint == null || endpoint.length() == 0) {
+      return true;
+    } else {
+      return endpoint.contains("https://");
```

Comment:
nitpick:
```suggestion
      return endpoint.startsWith("https://");
```

maybe we should be using a proper url parser, but that feels kind of overkill here